tests/dags/common/conftest.py

```diff
@@ -54,3 +60,77 @@ def identifier(request):
 def image_table(identifier):
     # Parallelized tests need to use distinct database tables
     return f"image_{identifier}"
+
+
+TEST_SQL = "SELECT PG_SLEEP(1);"
+
+
+def timed_pg_hook_sleeper(
+    task,
+    statement_timeout: float = None,
+):
+    pg = PostgresHook(
+        default_statement_timeout=PostgresHook.get_execution_timeout(task),
+        conn_id=POSTGRES_CONN_ID,
+    )
+    pg.run(sql=TEST_SQL, statement_timeout=statement_timeout)
+
+
+def mapped_select_pg_hook(
+    select_val: int,
+    task: BaseOperator,
+):
+    pg = PostgresHook(
+        default_statement_timeout=PostgresHook.get_execution_timeout(task),
+        conn_id=POSTGRES_CONN_ID,
+    )
+    return pg.run(f"select {select_val};")
+
+
+def create_pg_timeout_tester_dag():
+    with DAG(
+        dag_id="a_pg_timeout_tester",
+        schedule=None,
+        doc_md="DAG to test query timeouts in postgres",
+        start_date=datetime(2023, 1, 1),
+    ) as dag:
+        pg_operator_happy = PGExecuteQueryOperator(
+            task_id="pg_operator_happy",
+            retries=0,
+            conn_id=POSTGRES_CONN_ID,
+            sql=TEST_SQL,
+            execution_timeout=timedelta(seconds=2),
+            doc_md="Custom PG operator, with query finished before execution timeout",
+        )
+        pg_hook_happy = PythonOperator(
+            task_id="pg_hook_happy",
+            retries=0,
+            python_callable=timed_pg_hook_sleeper,
+            execution_timeout=timedelta(hours=2),
+            doc_md="Custom PG hook, with query finished before execution timeout",
+        )
+        pg_hook_no_timeout = PythonOperator(
+            task_id="pg_hook_no_timeout",
+            retries=0,
+            python_callable=timed_pg_hook_sleeper,
+            doc_md="Custom PG hook, with no execution timeout",
+        )
+        pg_operator_mapped = PythonOperator.partial(
+            task_id="pg_operator_mapped",
+            retries=0,
+            execution_timeout=timedelta(minutes=1),
+            doc_md="Custom PG operator, mapped to list",
+            python_callable=mapped_select_pg_hook,
+        ).expand(op_args=[(1,), (2,)])
+        [pg_operator_happy, pg_hook_happy, pg_hook_no_timeout, pg_operator_mapped]
+    return dag
+
+
+@pytest.fixture(scope="session")
+def mock_dag():
+    return create_pg_timeout_tester_dag()
+
+
+@pytest.fixture(scope="session")
+def mock_task(mock_dag) -> BaseOperator:
+    return mock_dag.get_task("pg_hook_happy")
```

Comment:
Since these fixtures are at almost the highest level `conftest.py` that means that they will be available to **all** tests in `tests/dags/common`. It might be prudent to have more specific names than `mock_dag` and `mock_task`, which speak to something much more generic than what's being described here and could be confusing for folks writing tests elsewhere. Maybe `mock_pg_timeout_dag` and `mock_pg_hook_task`?